ADR/0008-integration-service-api-contracts.md

```diff
@@ -0,0 +1,266 @@
+
+The Test Tekton tasks will use a standardized manner of displaying testing information. This information will be saved in the form of a JSON file. The contents of this file will relay results of validation by the [Open Policy Agent](https://www.openpolicyagent.org/) policies as executed by the [Conftest](https://www.conftest.dev/) tool and saved in its JSON output format.
+
+Each test will have a standardized short name in snake case, e.g. `release_label_required`, `architecture_label_required`, `architecture_label_deprecated` etc.
+
+The output will provide the following information about the overall test result:
+- **filename** - Name of the file that was inspected by the Tekton task
+  - This will reflect the data that the test was executed on, such as `image-inspect.json`, `clair-vulnerabilities.json` etc.
+- **namespace** - The rego namespace of the test policy. 
+- **successes** - The number of successful checks in the form of an integer
+- **failures** - A JSON list containing objects describing each failure
+
+The **failures** list of objects will provide the following information about each check:
+- **name** - Name of the individual check written in [snake case](https://en.wikipedia.org/wiki/Snake_case)
+- **msg** - Message about what went wrong with the check
+- **description** - The explanation about why the check is significant (Why the label is used, why the vulnerabilities are checked etc.)
+- **url** - link to the further documentation on the individual check (or that type of check)
+Example full JSON output for a Tekton task that tests the container image labels:
+```
+[
+    {
+        "filename": "image-inspect.json",
+        "namespace": "image_labels",
+        "successes": 19,
+        "failures": [
+            {
+                "msg": "The 'architecture' label is required",
+                "metadata": {
+                    "details": {
+                        "description": "Architecture the software in the image should target.",
+                        "name": "architecture_label_required",
+                        "url": "https://source.redhat.com/groups/public/container-build-system/container_build_system_wiki/guide_to_layered_image_build_service_osbs#jive_content_id_Labels"
+                    }
+                }
+            }
+          ]
+    }
+]
+```
+
+### Information injection
+
+Whenever possible, resources like ConfigMaps or Secrets will be used to inject configuration into Tasks. This is preferred to templates and patches as it fits well with Kubernetes declarative and GitOps approaches.
+
+ConfigMaps and Secrets will be mounted into the Task pod to inject file-based information like certificates.
+Environment variables may be injected from ConfigMaps.
+
+Variables directly configured in the Pod definition are discouraged.
+
+Side note: Secrets and ConfigMaps should be made [immutable](https://kubernetes.io/docs/concepts/configuration/secret/#secret-immutable)
+
+### Information vector
```

Comment:
I don't understand the purpose of this Information vector section.

There's a list of things, with no indication of when to use them. There's a (correct) note about the limmitations on task result sizes - which is actually more relevant for the section at the top of the Decision (it is why we need a minimal HACBS_TEST_OUTPUT). The sentence about the volume is also correct, but not sure why it is here.

Delete this section?

Reply:
Let's delete the list and move the limitation to the top of the `Decision`